firedrake/preconditioners/patch.py

```diff
@@ -772,15 +774,32 @@ def initialize(self, obj):
         if mesh.cell_set._extruded:
             raise NotImplementedError("Not implemented on extruded meshes")
 
-        if "overlap_type" not in mesh._distribution_parameters:
-            if mesh.comm.size > 1:
-                # Want to do
-                # warnings.warn("You almost surely want to set an overlap_type in your mesh's distribution_parameters.")
-                # but doesn't warn!
-                PETSc.Sys.Print("Warning: you almost surely want to set an overlap_type in your mesh's distribution_parameters.")
+        # Validate the mesh overlap
+        prefix = (obj.getOptionsPrefix() or "") + "patch_"
+        opts = PETSc.Options(prefix)
+        patch_dim = opts.getInt("pc_patch_construct_dim", default=0)
+        patch_type = opts.getString("pc_patch_construct_type", default="star")
+        patch_depth = {"pardecomp": 0, "star": 1, "vanka": 2}[patch_type]
+
+        tdim = self.plex.getDimension()
+        overlap_entity, overlap_depth = mesh._distribution_parameters["overlap_type"]
+        overlap_dim = {
+            DistributedMeshOverlapType.VERTEX: 0,
+            DistributedMeshOverlapType.FACET: tdim-1,
+            DistributedMeshOverlapType.NONE: tdim,
+        }[overlap_entity]
+
+        if mesh.comm.size > 1:
+            if overlap_dim > patch_dim:
+                patch_entity = {0: "vertex", 1: "edge", 2: "face", tdim: "cell"}[patch_dim]
```

Comment:
To be consistent with the `DistrubutedMeshOverlatType`
```suggestion
                patch_entity = {0: "vertex", 1: "edge", 2: "facet", tdim: "cell"}[patch_dim]
```

Reply:
We either refer to patch entities by their dimension (vertex, edge, face, cell) or by their codimension (cell, facet, ridge).

Reply:
if `tdim=2`, the dictionary reads `{0: "vertex", 1: "edge", 2: "cell"}`

Reply:
Similarly if `tdim=1`, the dictionary reads `{0: "vertex", 1: "cell"}` (we should ignore that there is `2: "face"` but it still makes sense)